airflow/providers/amazon/aws/operators/eks.py

```diff
@@ -0,0 +1,709 @@
+        self.region = region
+
+    def execute(self, context):
+        eks_hook = EKSHook(
+            aws_conn_id=self.conn_id,
+            region_name=self.region,
+        )
+
+        return eks_hook.list_nodegroups(clusterName=self.clusterName, verbose=self.verbose)
+
+
+class EKSPodOperator(KubernetesPodOperator):
+    """
+    Executes a task in a Kubernetes pod on the specified Amazon EKS Cluster.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:EKSPodOperator`
+
+    :param cluster_name: The name of the Amazon EKS Cluster to execute the task on.
+    :type cluster_name: str
+    :param cluster_role_arn: The Amazon Resource Name (ARN) of the IAM role that provides permissions
+       for the Kubernetes control plane to make calls to AWS API operations on your behalf.
+    :type cluster_role_arn: str
+    :param in_cluster: If True, look for config inside the cluster; if False look for a local file path.
+    :type in_cluster: bool
+    :param namespace: The namespace in which to execute the pod.
+    :type namespace: str
+    :param pod_context: The security context to use while executing the pod.
+    :type pod_context: str
+    :param pod_name: The unique name to give the pod.
+    :type pod_name: str
+    :param pod_username: The username to use while executing the pod.
+    :type pod_username: str
+    :param aws_profile: The named profile containing the credentials for the AWS CLI tool to use.
+    :param aws_profile: str
+    :param region: Which AWS region the connection should use.
+        If this is None or empty then the default boto3 behaviour is used.
+    :type region: str
+
+    """
+
+    def __init__(  # pylint: disable=too-many-arguments,too-many-locals
+        self,
+        cluster_name: str,
+        cluster_role_arn: Optional[str] = None,
+        # Setting in_cluster to False tells the pod that the config
+        # file is stored locally in the worker and not in the cluster.
+        in_cluster: Optional[bool] = False,
+        namespace: Optional[str] = DEFAULT_NAMESPACE_NAME,
+        pod_context: Optional[str] = DEFAULT_CONTEXT_NAME,
+        pod_name: Optional[str] = DEFAULT_POD_NAME,
+        pod_username: Optional[str] = DEFAULT_POD_USERNAME,
+        aws_profile: Optional[str] = None,
+        region: Optional[str] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(
+            in_cluster=in_cluster,
+            namespace=namespace,
+            name=pod_name,
+            **kwargs,
+        )
+        self.aws_profile = os.getenv('AWS_PROFILE', None) if aws_profile is None else aws_profile
+        self.roleArn = cluster_role_arn
+        self.clusterName = cluster_name
+        self.pod_name = pod_name
+        self.pod_username = pod_username
+        self.pod_context = pod_context
+        self.region = region
+
+    def execute(self, context):
+        self.config_file = generate_config_file(
```

Comment:
We should use the context manager here to ensure that the file is deleted when not in use.

Reply:
Should be addressed by https://github.com/apache/airflow/pull/16571/commits/8687b736994d1911011dd8bd2b2903d82ad30a8c